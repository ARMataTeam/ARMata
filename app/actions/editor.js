// @flow
export const SET_TEMPLATE = 'SET_TEMPLATE';
export const MARK_EDITED = 'MARK_EDITED';
export const MARK_IDLE = 'MARK_IDLE';
export const MARK_SAVED = 'MARK_SAVED';

type actionType = {
  type: string
};

export function setNewTemplate(template: string) {
  return (dispatch: (action: actionType) => void) => {
    dispatch({ type: SET_TEMPLATE, template });
  };
}

export function markEdited() {
  return (dispatch: (action: actionType) => void) => {
    dispatch({ type: MARK_EDITED });
  };
}

export function markIdle() {
  return (dispatch: (action: actionType) => void) => {
    dispatch({ type: MARK_IDLE });
  };
}

export function markSaved() {
  return (dispatch: (action: actionType) => void) => {
    dispatch({ type: MARK_SAVED });
  };
}
