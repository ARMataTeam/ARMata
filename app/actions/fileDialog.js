export const OPEN_FILE = 'OPEN_FILE';

export function open(selectedFilename) {
  return {
    type: OPEN_FILE, selectedFilename
  };
}
