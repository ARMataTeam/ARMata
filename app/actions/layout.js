// @flow
import { ProgressInfo } from 'electron-updater';

export const CHANGE_LAYOUT = 'CHANGE_LAYOUT';
export const CHANGE_VIEW = 'CHANGE_VIEW';
export const CLEAR_ERRORS = 'CLEAR_ERRORS';
export const ERROR = 'ERROR';
export const ALERT = 'ALERT';
export const PROGRESS = 'PROGRESS';
export const REPORT_ERROR = 'REPORT_ERROR';
export const OPEN_SETTINGS = 'OPEN_SETTINGS';
export const CLOSE_SETTINGS = 'CLOSE_SETTINGS';
export const OPEN_NODE_WINDOW = 'OPEN_NODE_WINDOW';
export const CLOSE_NODE_WINDOW = 'CLOSE_NODE_WINDOW';
export const OPEN_WINDOW = 'OPEN_WINDOW';
export const CLOSE_WINDOW = 'CLOSE_WINDOW';
export const OPEN_VISUALIZATION = 'OPEN_VISUALIZATION';
export const OPEN_TOOLBOX = 'OPEN_TOOLBOX';
export const CLOSE_TOOLBOX = 'CLOSE_TOOLBOX';
export const TOGGLE_PHYSICS = 'TOGGLE_PHYSICS';
export const ADD_RESOURCE = 'ADD_RESOURCE';
export const DELETE_RESOURCE = 'DELETE_RESOURCE';

type actionType = {
  type: string
};

type changeViewType = {
  type: string,
  view: string
};

export function toggleHierarchicalLayout() {
  return (dispatch: (action: actionType) => void) => {
    dispatch({ type: CHANGE_LAYOUT });
  };
}

export function togglePhysics() {
  return (dispatch: (action: actionType) => void) => {
    dispatch({ type: TOGGLE_PHYSICS });
  };
}

export function changeView(view: string) {
  return (dispatch: (action: changeViewType) => void) => {
    dispatch({ type: CHANGE_VIEW, view });
  };
}

export function dispatchButtonClick(action: string) {
  return (dispatch: (action: actionType) => void) => {
    dispatch({ type: action });
    dispatch({ type: CLEAR_ERRORS });
  };
}

export function openSettings() {
  return (dispatch: (action: actionType) => void) => {
    dispatch({ type: OPEN_SETTINGS });
  };
}

export function openNodeWindow(nodes: Array<string>) {
  return (dispatch: (action: actionType) => void) => {
    dispatch({ type: OPEN_NODE_WINDOW, nodes });
  };
}

export function openVisualization() {
  return (dispatch: (action: actionType) => void) => {
    dispatch({ type: OPEN_VISUALIZATION });
  };
}

export function openToolbox() {
  return (dispatch: (action: actionType) => void) => {
    dispatch({ type: OPEN_TOOLBOX });
  };
}

export function closeToolbox() {
  return (dispatch: (action: actionType) => void) => {
    dispatch({ type: CLOSE_TOOLBOX });
  };
}

export function error(errorMessage: string, title: string = 'Error occured') {
  const buttons = [{ label: 'Got it', action: CLEAR_ERRORS }, { label: 'Report', action: REPORT_ERROR }];

  return {
    type: ERROR, message: errorMessage, buttons, title
  };
}

export function alert(alertMessage: string, buttons: Object[] = []) {
  return {
    type: ALERT, message: alertMessage, buttons
  };
}

export function notifyProgress(progress: ProgressInfo) {
  return {
    type: PROGRESS, progress
  };
}

export function clearErrors() {
  return {
    type: CLEAR_ERRORS
  };
}

export function openWindow(name: string, title: string, content: string) {
  return {
    type: OPEN_WINDOW, windowName: name, window: { title, content }
  };
}

export function closeWindow() {
  return {
    type: CLOSE_WINDOW
  };
}

export function addResource(resourceType: string) {
  return (dispatch: (action: actionType) => void) => {
    dispatch({ type: ADD_RESOURCE, resourceType });
  };
}


export function deleteResource(id: string) {
  return (dispatch: (action: actionType) => void) => {
    dispatch({ type: DELETE_RESOURCE, id });
  };
}