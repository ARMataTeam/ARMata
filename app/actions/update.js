// @flow
export const CHECKING_FOR_UPDATE = 'CHECKINF_FOR_UPDATE';
export const UPDATE_AVAILABLE = 'UPDATE_AVAILABLE';
export const UPDATE_NOT_AVAILABLE = 'UPDATE_NOT_AVAILABLE';

type actionType = {
  type: string
};

export function checkingForUpdate() {
  return { type: CHECKING_FOR_UPDATE };
}

export function updateAvailable() {
  return { type: UPDATE_AVAILABLE };
}

export function updateNotAvailable() {
  return { type: UPDATE_NOT_AVAILABLE };
}
