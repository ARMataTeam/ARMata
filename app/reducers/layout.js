// @flow
import { ProgressInfo } from 'electron-builder-http';
import { CHANGE_LAYOUT, CHANGE_VIEW, CLEAR_ERRORS, ERROR, ALERT, PROGRESS, OPEN_SETTINGS, CLOSE_SETTINGS } from '../actions/layout';

type actionType = {
  type: string
};

type layoutStateType = {
  hierarchicalLayout: boolean,
  view: string,
  message: string,
  title: string,
  isError: boolean,
  buttons: Object[],
  progress: ProgressInfo,
  isSettingsWindowOpen: boolean
};

const initialState = {
  hierarchicalLayout: false,
  view: 'Structure',
  message: '',
  title: '',
  isError: false,
  buttons: [],
  progress: {},
  isSettingsWindowOpen: false
};

export default function layout(state: layoutStateType = initialState, action: actionType) {
  switch (action.type) {
    case CHANGE_LAYOUT:
      return Object.assign({}, state, {
        hierarchicalLayout: !state.hierarchicalLayout
      });
    case CHANGE_VIEW:
      return Object.assign({}, state, {
        view: action.view
      });
    case CLEAR_ERRORS:
      return Object.assign({}, state, {
        message: '',
        title: '',
        isError: false
      });
    case ERROR:
      return Object.assign({}, state, {
        message: action.message,
        title: action.title,
        isError: true,
        buttons: action.buttons
      });
    case ALERT:
      return Object.assign({}, state, {
        title: action.title,
        message: action.message,
        buttons: action.buttons
      });
    case PROGRESS:
      return Object.assign({}, state, {
        progress: action.progress
      });
    case OPEN_SETTINGS:
      return Object.assign({}, state, {
        isSettingsWindowOpen: true
      });
    case CLOSE_SETTINGS:
      return Object.assign({}, state, {
        isSettingsWindowOpen: false
      });
    default:
      return state;
  }
}
