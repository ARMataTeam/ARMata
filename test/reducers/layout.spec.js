import layout from '../../app/reducers/layout';
import { CHANGE_LAYOUT, CHANGE_VIEW, CLEAR_ERRORS } from '../../app/actions/layout';

describe('reducers', () => {
  describe('layout', () => {
    it('should handle initial state', () => {
      expect(layout(undefined, {})).toMatchSnapshot();
    });

    it('should handle CHANGE_LAYOUT', () => {
      expect(layout({
        hierarchicalLayout: false,
        view: 'Structure',
        message: '',
        isError: false
      }, { type: CHANGE_LAYOUT })).toMatchSnapshot();
    });

    it('should handle CHANGE_VIEW', () => {
      expect(layout({
        hierarchicalLayout: false,
        view: 'Structure',
        message: '',
        isError: false
      }, { type: CHANGE_VIEW, view: 'JsonPreview' })).toMatchSnapshot();
    });

    it('should handle CLEAR_ERRORS', () => {
      expect(layout({
        hierarchicalLayout: false,
        view: 'Structure',
        message: 'Some message',
        isError: true
      }, { type: CLEAR_ERRORS })).toMatchSnapshot();
    });
  });
});
