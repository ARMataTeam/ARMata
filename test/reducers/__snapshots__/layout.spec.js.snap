// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers layout should handle CHANGE_LAYOUT 1`] = `
Object {
  "hierarchicalLayout": true,
  "isError": false,
  "message": "",
  "view": "Structure",
}
`;

exports[`reducers layout should handle CHANGE_VIEW 1`] = `
Object {
  "hierarchicalLayout": false,
  "isError": false,
  "message": "",
  "view": "JsonPreview",
}
`;

exports[`reducers layout should handle CLEAR_ERRORS 1`] = `
Object {
  "hierarchicalLayout": false,
  "isError": false,
  "message": "",
  "title": "",
  "view": "Structure",
}
`;

exports[`reducers layout should handle initial state 1`] = `
Object {
  "buttons": Array [],
  "hierarchicalLayout": false,
  "isError": false,
  "isSettingsWindowOpen": false,
  "message": "",
  "progress": Object {},
  "title": "",
  "view": "Structure",
}
`;
